analysis/test_dm_reg_dataset_milan.py

```diff
@@ -0,0 +1,124 @@
+            {
+                "start_date": date(2010, 1, 1),
+            },
+        ],
+        "clinical_events": [
+            {
+                # First diabetes diagnosis (DM_COD)
+                "date": date(2024, 8, 1),
+                "snomedct_code": "73211009",
+            },
+        ],
+        "expected_in_population": False,
+    },
+    # Correctly not expected in population
+    # Younger than 17yo at index date
+    3: {
+        "patients": {"date_of_birth": date(2010, 1, 1)},
+        "practice_registrations": [
+            {
+                "start_date": date(2010, 1, 1),
+            },
+        ],
+        "clinical_events": [
+            {
+                # First diabetes diagnosis (DM_COD)
+                "date": date(2022, 8, 1),
+                "snomedct_code": "73211009",
+            },
+        ],
+        "expected_in_population": False,
+    },
+    # Correctly not expected in population
+    # Not registered at index date
+    4: {
+        "patients": {"date_of_birth": date(1960, 1, 1)},
+        "practice_registrations": [
+            {
+                "start_date": date(1960, 1, 1),
+                "end_date": date(2020, 1, 1),
+            },
+        ],
+        "clinical_events": [
+            {
+                # First diabetes diagnosis (DM_COD)
+                "date": date(2022, 8, 1),
+                "snomedct_code": "73211009",
+            },
+        ],
+        "expected_in_population": False,
+    },
+    # Correctly not expected in population
+    # Diabetes diagnosis resolved before index date
+    5: {
+        "patients": {"date_of_birth": date(1960, 1, 1)},
+        "practice_registrations": [
+            {
+                "start_date": date(1960, 1, 1),
+            },
+        ],
+        "clinical_events": [
+            {
+                # First diabetes diagnosis (DM_COD)
+                "date": date(2000, 6, 1),
+                "snomedct_code": "73211009",
+            },
+            {
+                # Diabetes diagnosis resolved (DMRES_COD)
+                "date": date(2023, 1, 1),
+                "snomedct_code": "315051004",
+            },
+        ],
+        "expected_in_population": False,
+    },
+    # Correctly expected in population
+    # Diabetes diagnosis resolved before index date
+    6: {
+        "patients": {"date_of_birth": date(1960, 1, 1)},
+        "practice_registrations": [
+            {
+                "start_date": date(1960, 1, 1),
+            },
+        ],
+        "clinical_events": [
+            {
+                # First diabetes diagnosis (DM_COD)
+                "date": date(2000, 6, 1),
+                "snomedct_code": "73211009",
+            },
+        ],
+        "expected_in_population": True,
+        "expected_columns": {
+            "pat_age": 64,
+            "dmlat_dat": date(2000, 6, 1),
+            "dmres_dat": None,
+            "dm_reg_r1": True,
+            "dm_reg_r2": False,
+        },
+    },
+}
```

Comment:
Would it be beneficial to include another patient test with an unrelated snomed code, to test the last_matching_event function's ability to match snomed codes with dm_cod codelist codes. 

```suggestion
       7: {
              "patients": {"date_of_birth": date(1960, 1, 1)},
              "practice_registrations": [
                  {
                      "start_date": date(1960, 1, 1),
                  },
              ],
              "clinical_events": [
                  {
                      # First diabetes diagnosis (DM_COD)
                      "date": date(2000, 6, 1),
                      "snomedct_code": "82272006",
                  },
              ],
              "expected_in_population": False,
          },
}
```